### PR TITLE
feature/ta sections

### PR DESCRIPTION
**Pull request by jamestjw**
- Assign existing TA to a section
- Create new user for TAs

Only instructors and TAs may comment on files and discussions

### CODE REVIEW

File: public/sections/create_new_ta.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+require_once "../../modules/models/section_ta.php";
+require_once "../../modules/models/section.php";
+require_once "../../common.php";
+
+$required_attrs = ["section_id", "first_name", "last_name", "password", "email"];
+
+foreach ($required_attrs as $attr) {
+    if (!array_key_exists($attr, $_POST)) {
+        die("Invalid $$attr");
```
Comment (chris7neves) [inline public/sections/create_new_ta.php:13]: double dollar sign is an accident here?

File: public/sections/create_new_ta.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    die("You must be an instructor of this course to modify a marked entity.");
+}
+
+// Check if the user is indeed an instructor of this course
+if (is_null(LectureInstructor::find_by(["lecture_id" => $section->lecture_id, "user_id"=> $_SESSION["current_user_id"]]))) {
+    die("Course instructor does not teach this course.");
+}
+
+// Check that the section has no TAs
+if (!is_null(SectionTA::find_by(["section_id" => $_POST["section_id"]]))) {
```
Comment (chris7neves) [inline public/sections/create_new_ta.php:31]: is there a way to modify if someone is a ta or not?

File: public/view_lecture_tas.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+} catch (PDOException $error) {
+    echo "<br>" . $error->getMessage();
+}
+
+?> 
+<h2>Teaching Assistants</h2>
+<?php
+
+foreach ($course_sections as $course_section) {
+    echo "<h4>{$course_section->lecture->lecture_code} - {$course_section->section_code}</h3>";
```
Comment (chris7neves) [inline public/view_lecture_tas.php:22]: opening with h4 closing with h3 here

File: public/view_lecture_tas.php [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            <option value = "" disabled selected>--Select section--</option>
+            <?php foreach ($sections_without_tas as $section):; ?>
+                <option value = <?php echo($section->id); ?>><?php echo "{$section->lecture->lecture_code} - {$section->section_code}"; ?></option>
+            <?php endforeach; ?>
+        </select>
+    </div>
+
+    <div>
+        <div>
+            <label for="first_name">First Name</label>
```
Comment (chris7neves) [inline public/view_lecture_tas.php:74]: should all these fields be made required?

**Review by chris7neves — CHANGES_REQUESTED** [review]

File: public/sections/create_new_ta.php (hunk 1 above)
Comment (jamestjw) [reply public/sections/create_new_ta.php:13]: ok good catch

File: public/sections/create_new_ta.php (hunk 2 above)
Comment (jamestjw) [reply public/sections/create_new_ta.php:31]: nope I guess not 😅 maybe if we have time we can allow the admin to change all these little things?

File: public/view_lecture_tas.php (hunk 3 above)
Comment (jamestjw) [reply public/view_lecture_tas.php:22]: nice catch not sure how you spotted that tho 😂

File: public/view_lecture_tas.php (hunk 4 above)
Comment (jamestjw) [reply public/view_lecture_tas.php:74]: yes for sure

**Comment by jamestjw** [comment]
@chris7neves okay I addressed your comments!

**Comment by jamestjw** [comment]
sec, finishing this up ..

**Comment by jamestjw** [comment]
done

**Pull request merged**